geode-core/src/distributedTest/java/org/apache/geode/cache30/CacheStatisticsPartitionedRegionDUnitTest.java

```diff
@@ -0,0 +1,464 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache30;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import org.apache.geode.cache.AttributesFactory;
+import org.apache.geode.cache.CacheException;
+import org.apache.geode.cache.CacheStatistics;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.StatisticsDisabledException;
+import org.apache.geode.internal.cache.InternalRegion;
+import org.apache.geode.test.dunit.SerializableRunnableIF;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.cache.internal.JUnit4CacheTestCase;
+
+/**
+ * Tests the {@link CacheStatistics} that are maintained by a partitioned {@link Region} .
+ *
+ *
+ * @since GemFire 3.0
+ */
+
+public class CacheStatisticsPartitionedRegionDUnitTest extends JUnit4CacheTestCase {
```

Comment:
One other thing I think this test needs to cover is a PartitionedRegion that is a  "proxy". You can get one of these by creating the PartitionedRegion with the RegionShortcut.PARTITION_PROXY or by using PartitionAttributesFactory().setLocalMaxMemory(0).

I think it would be okay for a proxy to just return 0 for all these stats. The test could just verify the current behavior of these apis on a partitioned proxy and the javadocs on Region should describe that behavior.

Reply:
You are right. I forgot about that.

Reply:
I have pushed two new commits: one with all the new comments except the one for the PARTITION_PROXY test and another one with this test.